### PR TITLE
One more batch of API deprecations

### PR DESCRIPTION
**Pull request by kgaillot**
This side project turned out to be something of a time sink, so this is the last batch planned for now. The most important thing here is fixing two Y2038 bugs.

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens, can you review this (thankfully short) PR when you get a chance?

File: daemons/execd/pacemaker-execd.c [hunk 1]
```diff
@@ -487,8 +487,15 @@ main(int argc, char **argv, char **envp)
...
-    /* Used by RAs - Leave owned by root */
-    crm_build_path(CRM_RSCTMP_DIR, 0755);
+    {
+        // Temporary directory for resource agent use (leave owned by root)
+        int rc = pcmk__build_path(CRM_RSCTMP_DIR, 0755);
+
+        if (rc != pcmk_rc_ok) {
+            crm_warn("Could not create resource agent temporary directory "
+                     CRM_RSCTMP_DIR ": %s", pcmk_rc_str(rc));
+        }
```
Comment (clumens) [inline daemons/execd/pacemaker-execd.c:497]: If this fails, it's okay for pacemaker-execd to continue?

**Comment by clumens** [comment]
This all looks good to me.

File: daemons/execd/pacemaker-execd.c (hunk 1 above)
Comment (kgaillot) [reply daemons/execd/pacemaker-execd.c:497]: Yes, this is not used by Pacemaker itself, but by some resource agents. If it doesn't exist, they can provide error reporting/recovery as desired, without affecting the cluster itself or agents that don't use it.

**Pull request merged**